src/Components/NewCommentForm.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState, useEffect } from "react";
+import { getAllUsers } from "../utils/getAllUsers";
+import { postComment } from "../utils/postComment";
+import { useParams } from "react-router-dom";
+
+const NewCommentForm = ({ setComments }) => {
+	const [isLoading, setIsLoading] = useState(true);
+	const [users, setUsers] = useState({});
+	const [newComment, setNewComment] = useState("");
+	const [selectedUser, setSelectedUser] = useState("");
+	const [submitted, setSubmitted] = useState(false);
+	const { article_id } = useParams();
+
+	useEffect(() => {
+		setIsLoading(true);
+		getAllUsers()
+			.then((data) => {
+				setUsers(data.users);
+			})
+			.then(() => {
+				setTimeout(() => {
+					setIsLoading(false);
+				}, 2000);
+			});
+	}, []);
+
+	const handleSubmit = (event) => {
+		event.preventDefault();
+		setSubmitted(true);
+		const body = {
+			username: selectedUser,
+			post: newComment,
+		};
+		postComment(article_id, body.username, body.post)
```

Comment:
why not send the whole body here and directly put this object into you axios request rather than separating them and rebuilding the object again inside postComments?